src/pages/OrderConfirmPage.tsx

```diff
@@ -1,25 +1,103 @@
-import { useLocation } from 'react-router';
+import { useState } from 'react';
 import Header from '../components/Header/Header';
-import styled from '@emotion/styled';
+import CartInfo from '../components/Cart/CartInfo';
+import CartHeader from '../components/Cart/CartHeader';
+import CartOrderItem from '../components/Cart/CartOrderItem';
+import DeliveryOptions from '../components/Cart/DeliveryOptions';
+import CartFooter from '../components/Cart/CartFooter';
 import Button from '../components/Button/Button';
+import CouponModal from '../components/Modal/CouponModal';
+import styled from '@emotion/styled';
+import { css } from '@emotion/react';
+import { useLocation, useNavigate } from 'react-router';
+import { SHIPPING_FEE_THRESHOLD } from '../constants/cartConfig';
+import { useOrderConfirm } from '../hooks/useOrderConfirm';
+import { Coupon } from '../types/coupon';
 
 function OrderConfirmPage() {
   const location = useLocation();
-  const { price, count, totalCount } = location.state;
+  const navigate = useNavigate();
+  const { checkedItemIds, price, count, totalCount } = location.state || {};
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const {
+    selectedCoupons,
+    setSelectedCoupons,
+    remoteArea,
+    toggleRemoteArea,
+    totalShippingFee,
+    baseShippingFee,
+    remoteAreaFee,
+    couponDiscount,
+    finalTotal,
+    products: orderProducts,
+    price: orderPrice,
+  } = useOrderConfirm();
+
+  const handleModalOpen = () => {
+    setIsModalOpen(true);
+  };
+
+  const handleModalClose = () => {
+    setIsModalOpen(false);
+  };
+
+  const handleConfirmCoupon = (coupons: Coupon[]) => {
+    setSelectedCoupons(coupons);
+    setIsModalOpen(false);
+  };
```

Comment:
```suggestion
  const openModal = () => {
    setIsModalOpen(true);
  };

  const closeModal = () => {
    setIsModalOpen(false);
  };

  const applyCopuons = (coupons: Coupon[]) => {
    setSelectedCoupons(coupons);
    setIsModalOpen(false);
  };
```

함수가 하는 일로 이름을 지어주세요!